hash_practice/exercises.py

```diff
@@ -1,29 +1,131 @@
-
-def grouped_anagrams(strings):
-    """ This method will return an array of arrays.
+""" This method will return an array of arrays.
         Each subarray will have strings which are anagrams of each other
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n log n)
+        Space Complexity: O(n)
     """
-    pass
+def grouped_anagrams(strings):
+    map = {}
+    for string in strings:
+        sort = str(''.join(sorted(string)))
+        if sort not in map: 
+            map[sort] = [string]
+        else:
+            list = map[sort]
+            list.append(string)
+            map[sort] = list
+    
+    lists = []
+    for key in map:
+        lists.append(map[key])
+    return lists
+
+
+def makeMap(string):
+    map = {}
+
+    for char in string:
+        if not map.get(char):
+            map[char] = 1
+        else:
+            map[char] += 1
+    
+    return map
 
 def top_k_frequent_elements(nums, k):
     """ This method will return the k most common elements
         In the case of a tie it will select the first occuring element.
-        Time Complexity: ?
-        Space Complexity: ?
-    """
-    pass
+        Time Complexity: O(n log n)
+        Space Complexity: O(n)
+    """    
+    if not nums:
+        return []
+
+    map = makeMap(nums)

```

Comment:
In general try to make sure to use more descriptive names for your variables. It's generally less important _what_ a variable is and more important _why_ a variable exists. So rather than a "what" name like `map`, I would probably use something like `number_frequency_map` or something like that.

This seems to be general to the code here, so I'm only going to mention it once.